### ISSUE TITLE
Recommender: InvertedIndex returns a little wrong value

### ISSUE THREAD

**Issue by rimms** [jubatus/jubatus_core#251]
My test code is followings:
```py
#!/usr/bin/env python

host = '127.0.0.1'
port = 9199
name = 'test'

import jubatus
from jubatus.common import Datum

if __name__ == '__main__':
    client = jubatus.Recommender(host, port, name)

    d = Datum()
    for i in range(30):
        d.add_number('key_{0:03d}'.format(i), 1.0)

    client.update_row('id', d)
    results = client.similar_row_from_datum(d, 1)

    print 'score = {0}'.format(results[0].score)
```

config:
```js
{
  "converter" : {
    "num_rules": [
      {"key" : "*", "type" : "num"}
    ]
  },
  "method": "inverted_index"
}
```

Expected: 1.0
Actual: 0.999999940395

**Comment by rimms** [comment]
Test with ``inverted_index_euclid``:

Expected: 0
Actual: -0.001953125

**Comment by kmaehashi** [comment]
I think we may solve this problem by changing the Distance(X,Y) function to behave as:

* If `N^2(X) == N^2(Y) == inp(X,Y)` then return `0.0`.
* Otherwise return `inp(X,Y) / (N(X) * N(Y))`. This is the current behavior.

where N^2(X) = squared L2-norm of vector X, N(X) = L2-norm of vector X, inp(X,Y) = inner product of vector X and Y.

**Comment by kmaehashi** [comment]
I noticed that in Inverted Index Euclid, operations that can be eliminated (e.g., `sqrt(x) * sqrt(x)`) drops the floating point precision.  I think it's worth fixing.

**Comment by kmaehashi** [comment]
Fixed via #290
